chapters/publication.tex

```diff
@@ -99,91 +181,265 @@ \section{Collaborating on academic writing}
+(for example, in Stata using \texttt{ssc install} or \texttt{net install}
+and in R include code giving users the option to install packages,
+including selecting a specific version of the package if necessary).
+In many cases you can even directly provide the underlying code
+for any user-installed packages that are needed to ensure forward-compatibility.
+Make sure system settings like \texttt{version}, \texttt{matsize}, and \texttt{varabbrev} are set.
+
+Finally, make sure that the code and its inputs and outputs are clearly identified.
+A new user should, for example, be able to easily identify and remove
+any files created by the code so that they can be recreated quickly.
+They should also be able to quickly map all the outputs of the code
+to the locations where they are placed in the associated published material,
+such as ensuring that the raw components of figures or tables are clearly identified.
+For example, outputs should clearly correspond by name to an exhibit in the paper, and vice versa.
+(Supplying a compiling \LaTeX\ document can support this.)
+Code and outputs which are not used should be removed.
+
+\subsection{Releasing a replication package}
 
 If you are at this stage,
-all you need to do is find a place to publish your work!
-GitHub provides one of the easiest solutions here,
-since it is completely free for static, public projects
-and it is straightforward to simply upload a fixed directory
-and obtain a permanent URL for it.
-The Open Science Framework also provides a good resource,
-as does ResearchGate (which can also assign a permanent
-digital object identifier link for your work).
+all you need to do is find a place to publish your materials.
+This is slightly easier said than done,
+as there are a few variables to take into consideration
+and, at the time of writing, no global consensus on the best solution.
+The technologies available are likely to change dramatically
+over the next few years;
+the specific solutions we mention here highlight some current approaches
+as well as their strengths and weaknesses.
+GitHub provides one solution.
+Making a GitHub repository public is completely free.
+It can hold any file types,
+provide a structured download of your whole project,
+and allow others to look at alternate versions or histories easily.
+It is straightforward to simply upload a fixed directory to GitHub
+apply a sharing license, and obtain a URL for the whole package.
+(However, there is a strict size restriction of 100MB per file and
+a restriction on the size of the repository as a whole,
+so larger projects will need alternative solutions.)
+
+However, GitHub is not ideal for other reasons.
+It is not built to hold data in an efficient way
+or to manage licenses or citations for datasets.
+It does not provide a true archive service --
+you can change or remove the contents at any time.
+A repository such as the Harvard Dataverse\sidenote{
```

Comment:
Is GitHub our first recommendation, given the drawbacks? are there drawbacks of dataverse?

Reply:
GitHub is nice but proprietary, but since MS owns it it's like Office now.... data verse is not good for code last I checked as it didn't preserve directory structures well but maybe that's changed?

Reply:
in #350